subprojects/docs/src/docs/userguide/jvm/java_plugin.adoc

```diff
@@ -515,6 +515,20 @@ To help you understand how incremental compilation works, the following provides
 * Having a source structure that does not match the package names, while legal for compilation, might end up causing trouble in the toolchain.
     Even more if annotation processing and <<build_cache.adoc#build_cache,caching>> are involved.
 
+Additionally, Gradle might temporary change the output location while running incremental compilation.
+This might affect some annotation processors that inspect output locations and which allow users to wire some action to file paths (e.g. `-XepExcludedPaths` in Error Prone).
+You can disable this Gradle behaviour by setting link:{javadocPath}/org/gradle/api/tasks/compile/CompileOptions.html#getIncrementalAfterFailure--[`options.incrementalAfterFailure`] to `false` on the JavaCompile task.
+Alternatively, you can try to specify a temporary output path to the annotation processor.
```

Comment:
I'm not sure I understand this sentence properly. Is there some mechanism for users to configure what the temporary output path is? Or is this a more general comment for annotation processors which themselves can be configured with custom file paths?

Reply:
Thanks for the review, I believe this might not be clear to a lot of users.

Users can't change that temporary output, but they can make annotation aware of it.
So basically what I want to say here is: 
The user has two options:
1.) They can just disable the behavior, done.
2.) Or they pass this value to the annotation processor as a parameter, so the annotation processor is aware of that path. In case of Error Prone they can pass that value to `-XepExcludedPaths`. Example:
Before they had: `-XepExcludedPaths=.*/build/generated/.*"`. With new behaviour that might not be good, so they should add `-XepExcludedPaths=.*/(build/generated|<task-name>/compileTransaction/annotation-output)/.*"` (tbh, l hope in Gradle 8.1 we get rid of that workaround with https://github.com/gradle/gradle/pull/23226).

Reply:
This extra explanation is nice, are you adding more into the doc for the users?

Reply:
Perhaps: 

```
This might affect some annotation processors that inspect output locations or
accept file paths as arguments. For example, Error Prone's `-XepExcludedPaths` 
parameter can be updated to `.*/(build/generated|compileJava/compileTransaction/annotation-output)/.*` 
in order to make the processor aware of the new temporary directory. 
```